packages/common/src/types.ts

```diff
@@ -27,7 +27,42 @@ export type NetworkId = Opaque<string, 'NetworkId'>;
 export type NetworkState = Opaque<string, 'NetworkState'>;
 
 export type BlockDetails = [BlockNumber, BlockHash, Milliseconds, Timestamp, Maybe<PropagationTime>];
-export type NodeDetails = [NodeName, NodeImplementation, NodeVersion, Maybe<Address>, Maybe<NetworkId>];
+export type NodeDetails = [NodeName, Address, NodeImplementation, NodeVersion, Maybe<Address>, Maybe<NetworkId>];
 export type NodeStats = [PeerCount, TransactionCount];
 export type NodeHardware = [Array<MemoryUse>, Array<CPUUse>, Array<BytesPerSecond>, Array<BytesPerSecond>, Array<Timestamp>];
 export type NodeLocation = [Latitude, Longitude, City];
+
+export declare type Authority = {
+  Address: Address,
+  NodeId: Maybe<NodeId>,
+  Name: Maybe<NodeName>,
+};
+export declare type Authorities = Array<Address>;
+export declare type AuthoritySetId = Opaque<number, 'AuthoritySetId'>;
+export declare type ConsensusInfo = {
+  BlockNumber: ConsensusView;
```

Comment:
I assume this is meant to be mapping. This creates a type definition for an object literal with a field name `BlockNumber`, not a mapping of block numbers to views.

In general, I prefer to use [`Map`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) for mappings, it has some advantages over object literals (most importantly, it allows for keys to be any values, it does not cast them to strings in order to hash them). So this would be:

`export type ConsensusInfo = Map<BlockNumber, ConsensusView>;`

That said mappings like these aren't very good candidates for being transmitted over the wire, as they can grow quite large. I'd prefer doing similar thing to what I'm doing with nodes: there are separate states in backend and frontend, and the backend is sending updates made to the state to the frontend without sending the whole thing over.

Reply:
> This creates a type definition for an object literal with a field name BlockNumber, not a mapping of block numbers to views.

Ahh, I see.

> […] the backend is sending updates made to the state to the frontend without sending the whole thing over

That's also the way it works in this PR (https://github.com/paritytech/substrate-telemetry/pull/134/files#diff-4e01c1b28b795b06884aafdfd8e1caebR199) ‒ only the blocks for which the consensus info changed since the last broadcast are propagated to the frontend. 

> mappings like these aren't very good candidates for being transmitted over the wire, as they can grow quite large.

This PR keeps only a fixed amount of blocks in the map (https://github.com/paritytech/substrate-telemetry/pull/134/files#diff-4e01c1b28b795b06884aafdfd8e1caebR10), hence it grows only to a fixed size.

> I'm happy to help implementing the changes, but it will have to wait till after Sub0

:+1: